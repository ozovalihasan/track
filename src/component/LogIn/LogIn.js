import { Redirect } from 'react-router-dom';
import { useSelector } from 'react-redux';
import styled from 'styled-components';

import { fetchUserLogin } from '../../redux';
import UserFormContainer from '../UserForm/UserFormContainer';
import LogInAsGuestButtonContainer from '../LogInAsGuestButton/LogInAsGuestButtonContainer';

const LogIn = () => {
  const { username } = useSelector(state => state.user);

  return username ? (
    <Redirect to={{ pathname: '/' }} />
  ) : (
    <Profile>
      <UserFormContainer fetchUser={fetchUserLogin} buttonName="Log In" />
      <LogInAsGuestButtonContainer />
    </Profile>
  );
};

const Profile = styled.div`
  height: 100vh;
`;

export default LogIn;
