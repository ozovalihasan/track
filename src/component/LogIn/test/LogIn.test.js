import { render, screen } from '@testing-library/react';
import React from 'react';
import {
  BrowserRouter, Redirect, Route, Switch,
} from 'react-router-dom';
import '@testing-library/jest-dom';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import LogIn from '../LogIn';

jest.mock('../../UserForm/UserFormContainer', () => {
  // eslint-disable-next-line global-require
  const PropTypes = require('prop-types');
  const UserFormContainer = ({ fetchUser, buttonName }) => (
    <div>
      {buttonName}
      {fetchUser.name}
      Mock UserFormContainer
    </div>
  );
  UserFormContainer.displayName = 'UserFormContainer';
  UserFormContainer.propTypes = {
    buttonName: PropTypes.string.isRequired,
    fetchUser: PropTypes.func.isRequired,
  };

  return UserFormContainer;
});

jest.mock('../../LogInAsGuestButton/LogInAsGuestButtonContainer', () => {
  const LogInAsGuestButtonContainer = () => (<div>Mock LogInAsGuestButtonContainer </div>);
  LogInAsGuestButtonContainer.displayName = 'LogInAsGuestButtonContainer';
  return LogInAsGuestButtonContainer;
});


const initStore = { user: { } };
const mockStore = configureStore();
const store = mockStore(initStore);

let renderReadyComponent;
beforeEach(() => {
  renderReadyComponent = (
    <BrowserRouter>
      <Provider store={store}>
        <Redirect to="/second-page" />
        <LogIn />
        <Switch>
          <Route exact path="/" render={() => <>Main Page</>} />
          <Route exact path="/second-page" render={() => <div>Second Page</div>} />
        </Switch>
      </Provider>
    </BrowserRouter>
  );
});

describe('<LogIn />', () => {
  describe('If username exists', () => {
    it('redirect to \'/\' ', () => {
      initStore.user.username = 'mockName';
      render(renderReadyComponent);

      expect(screen.getByText(/Main Page/i)).toBeInTheDocument();
      expect(screen.queryByText(/Second Page/i)).not.toBeInTheDocument();
    });

    it('renders correctly', () => {
      initStore.user.username = 'mockName';
      const renderedContainer = render(renderReadyComponent);
      expect(renderedContainer).toMatchSnapshot();
    });
  });

  describe('If username does not exist', () => {
    it('renders UserFormContainer if username is empty', () => {
      initStore.user.username = '';
      render(renderReadyComponent);

      expect(screen.getByText(/Mock UserFormContainer/i)).toBeInTheDocument();
      expect(screen.getByText(/Mock LogInAsGuestButtonContainer/i)).toBeInTheDocument();
      expect(screen.getByText(/Second Page/i)).toBeInTheDocument();
      expect(screen.getByText(/Log In/i)).toBeInTheDocument();
      expect(screen.getByText(/fetchUserLogin/i)).toBeInTheDocument();
      expect(screen.queryByText(/Main Page/i)).not.toBeInTheDocument();
    });

    it('renders correctly', () => {
      initStore.user.username = '';
      const renderedContainer = render(renderReadyComponent);

      expect(renderedContainer).toMatchSnapshot();
    });
  });
});
