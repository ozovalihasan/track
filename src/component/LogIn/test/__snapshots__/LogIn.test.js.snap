// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogIn /> If username does not exist renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdfBwQ bIQWAJ"
      >
        <div>
          Log in
          fetchUserLogin
          Mock UserFormContainer
        </div>
        <div>
          Mock LogInAsGuestButtonContainer 
        </div>
      </div>
      <div>
        Second Page
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdfBwQ bIQWAJ"
    >
      <div>
        Log in
        fetchUserLogin
        Mock UserFormContainer
      </div>
      <div>
        Mock LogInAsGuestButtonContainer 
      </div>
    </div>
    <div>
      Second Page
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<LogIn /> If username exists renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      Main Page
    </div>
  </body>,
  "container": <div>
    Main Page
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
